ignite/contrib/handlers/time_profilers.py

```diff
@@ -437,3 +438,395 @@ def odict_to_str(d: Mapping):
         )
         print(output_message)
         return output_message
+
+
+class HandlersTimeProfiler:
+    """
+    HandlersTimeProfiler can be used to profile the handlers,
+    data loading and data processing times.
+
+    Examples:
+
+    .. code-block:: python
+
+        from ignite.contrib.handlers import HandlersTimeProfiler
+
+        trainer = Engine(train_updater)
+
+        # Create an object of the profiler and attach an engine to it
+        profiler = HandlersTimeProfiler()
+        profiler.attach(trainer)
+
+        @trainer.on(Events.EPOCH_COMPLETED)
+        def log_intermediate_results():
+            profiler.print_results(profiler.get_results())
+
+        trainer.run(dataloader, max_epochs=3)
+
+        profiler.write_results('path_to_dir/time_profiling.csv')
+
+    """
+
+    def __init__(self):
+        self._dataflow_timer = Timer()
+        self._processing_timer = Timer()
+        self._event_handlers_timer = Timer()
+
+        self.dataflow_times = None
+        self.processing_times = None
+        self.event_handlers_times = None
+
+        # Events to profile
+        self._events = [
+            Events.STARTED,
+            Events.EPOCH_STARTED,
+            Events.EPOCH_COMPLETED,
+            Events.ITERATION_STARTED,
+            Events.ITERATION_COMPLETED,
+            Events.GET_BATCH_STARTED,
+            Events.GET_BATCH_COMPLETED,
+            Events.COMPLETED,
+        ]
+
+        # Events which are triggered only once
+        self._global_events = [Events.STARTED, Events.COMPLETED]
```

Comment:
I'm not sure to understand why we need to split by type of event like that : global, epoch, iteration. Can't we treat all events in the same way as we just need to measure the time passed inside a handler.

Reply:
I did that mostly because we don't know which handler will get triggered how many times. So this split allows me to statically initialize torch.tensor for every event measurements. We can use list and remove this split by type of event. But I was worried about python's list performance and memory because handlers can get triggered many times depending on number of iterations or trigger conditions.

Anyway, currently I am removing this splitting and making it more general to support custom events as well. But I believe we would still like to use tensor for storing these measurements. Let me know if you have any idea or suggestion on that.